python/lsst/validate/drp/matchreduce.py

```diff
@@ -345,101 +343,142 @@ def _loadAndMatchCatalogs(repo, dataIds, matchRadius,
         if len(cat) < nMatchesRequired:
             return False
-        for flagKey in flagKeys:
+        for flagKey in keys.flags:
             if cat.get(flagKey).any():
                 return False
-        if not np.isfinite(cat.get(magKey)).all():
+        if not np.isfinite(cat.get(keys.mag)).all():
             return False
-        extendedness = cat.get(extendedKey)
+        extendedness = cat.get(keys.extended)
         return np.min(extendedness) >= safeExtendedness if extended else \
             np.max(extendedness) < safeExtendedness
 
     def snrFilter(cat):
         # Note that this also implicitly checks for psfSnr being non-nan.
-        snr = np.median(cat.get(snrKey))
+        snr = np.median(cat.get(keys.snr))
         return snrMax >= snr >= snrMin
 
     def fullFilter(cat):
         return extendedFilter(cat) and snrFilter(cat)
 
-    # If safeSnr range is a subset of goodSnr, it's safe to only filter on snr again
+    # If brightSnr range is a subset of faintSnr, it's safe to only filter on snr again
     # Otherwise, filter on flags/extendedness first, then snr
-    isSafeSubset = goodSnrMax >= safeSnrMax and goodSnr <= safeSnr
-    goodMatches = allMatches.where(fullFilter) if isSafeSubset else allMatches.where(extendedFilter)
-    snrMin, snrMax = safeSnr, safeSnrMax
-    safeMatches = goodMatches.where(snrFilter)
+    isSafeSubset = faintSnrMax >= brightSnrMax and faintSnr <= brightSnr
+    matchesFaint = allMatches.where(fullFilter) if isSafeSubset else allMatches.where(extendedFilter)
+    snrMin, snrMax = brightSnr, brightSnrMax
+    matchesBright = matchesFaint.where(snrFilter)
+    # This means that matchesFaint has had extendedFilter but not snrFilter applied
     if not isSafeSubset:
-        snrMin, snrMax = goodSnr, goodSnrMax
-        goodMatches = goodMatches.where(snrFilter)
+        snrMin, snrMax = faintSnr, faintSnrMax
+        matchesFaint = matchesFaint.where(snrFilter)
+
+    return pipeBase.Struct(
+        extended=extended, keys=keys, matchesFaint=matchesFaint, matchesBright=matchesBright,
+    )
+
 
+def summarizeSources(blob, filterResult):
+    """Calculate summary statistics for each source. These are persisted
+    as object attributes.
+
+    Parameters
+    ----------
+    blob : `lsst.verify.blob.Blob`
+        A verification blob to store Datums in.
+    filterResult : `lsst.pipe.base.Struct`
+        A struct containing bright and faint filter matches, as returned by `filterSources`.
+    """
     # Pass field=psfMagKey so np.mean just gets that as its input
-    typeMag = "model" if extended else "PSF"
+    typeMag = "model" if filterResult.extended else "PSF"
     filter_name = blob['filterName']
-    source_type = f'{"extended" if extended else "point"} sources"'
-    blob['snr'] = Datum(quantity=goodMatches.aggregate(np.median, field=snrKey) * u.Unit(''),
+    source_type = f'{"extended" if filterResult.extended else "point"} sources"'
+    matches = filterResult.matchesFaint
```

Comment:
It's not clear why you are selecting the faint matches here. Does summarizeSources only aggregate the faint sources? (If so, perhaps it should be renamed?)

Reply:
Some of the astrometry metrics use 'snr' and filter on a different brightSnr parameter, whereas the photometry ones just use what used to be called safeMatches. Yes, it's confusing; I didn't want to tackle all of that in this ticket.

Reply:
Fair enough. We can clarify that later.